test/functional/test_framework/wallet.py

```diff
@@ -27,12 +31,20 @@
 
 
 class MiniWallet:
-    def __init__(self, test_node, *, raw_script=False):
+    def __init__(self, test_node, *, raw_script=False, use_p2pk=False):
         self._test_node = test_node
         self._utxos = []
+        self._priv_key = None
         if raw_script:
             self._address = None
-            self._scriptPubKey = bytes(CScript([OP_TRUE]))
+            if use_p2pk:
+                # use simple deterministic private key (k=1)
+                self._priv_key = ECKey()
+                self._priv_key.set((1).to_bytes(32, 'big'), True)
+                pub_key = self._priv_key.get_pubkey()
+                self._scriptPubKey = bytes(CScript([pub_key.get_bytes(), OP_CHECKSIG]))
+            else:
+                self._scriptPubKey = bytes(CScript([OP_TRUE]))
         else:
```

Comment:
would be nice to make this an `elif use_p2pk` to make it easier to switch to an enum

Reply:
Thanks, done.